docs/roadmap.md

```diff
@@ -0,0 +1,170 @@
+# Roadmap
+
+* [**Simple installation**](#simple-installation): importing a project should be
+  simple and require as few steps as possible.
+* [**Correct diagnostics**](#correct-diagnostics): the editor should not show
+  spurious red squiggles, even at the cost of higher latency.
+* [**Robust navigation**](#robust-navigation): goto definition should work
+  everywhere, even for Java dependencies.
+* [**Fast completions**](#fast-completions): completions suggestions should
+  respond in ~200ms, even at the cost of occasionally wrong results.
+* [**Low CPU and memory usage**](#low-cpu-and-memory-usage): indexing should run
+  in the background and not get in your way from coding.
+* [**Solid refactoring**](#solid-refactoring): rename, organize imports and
+  insert type annotation should always leave the codebase in a compiling state.
+
+## Simple installation
+
+The first you do with an IDE is boring but critical: import a project. The
+Language Server Protocol does not provide utilities to extract metadata from a
+build tool (module source files, dependencies, configuration data), so we are
+still exploring what is the best option.
+
+* [ ] sbt-metals: plugin that provides a `*:metalsSetup` task to export sbt
+      build metadata.
+* [ ] [BSP][]: LSP-inspired protocol to standardize on communication between a
+      language server and build tool. In theory, should enable automatic
+      importing of projects without custom build plugins or manual installation
+      steps.
+
+## Correct diagnostics
+
+Red squiggles are one of the most basic but also difficult-to-support features
+in an IDE. The Scala compiler was not designed for interactivity, which makes
+fast and correct diagnostics difficult to support in the editor. We are still
+unsure what the best model is for diagnostics, so we are still exploring what is
+the best option.
+
+* [x] Scala Presentation Compiler: interactively publish squiggles as you type.
+      Provides low latency but may get off-sync with build diagnostics.
+  * Disabled by default. Can be enabled with configuration option
+    `scalac.diagnostics.enabled=true`.
+  * Requires [`metalsSetup`](#metalssetup).
+* [x] sbt server: sbt 1.1 added LSP support so that build errors are reported
+      directly to the editor. Metals can connect to sbt server as a client to
+      trigger batch compilation on file save. Guarantees diagnostics are always
+      in-sync with build diagnostics but at the cost of higher latency.
+  * Disabled by default. Can be enabled with configuration option
+    `sbt.enabled=true`.
+  * Requires sbt 1.1 or later.
```

Comment:
I would add that it requires launching sbt manually to motivate more why it's an opt-in. I also don't expect that it will change anytime soon (say sbt 1.2).

Reply:
I added "  * Requires a running sbt 1.1 server or later."